specs/ServerCertificate.md

```diff
@@ -0,0 +1,229 @@
+Server Certificate API
+===
+# Background
+
+The WebView2 team has been asked for an API to intercept when WebView2 notifies a user about the risk of loading a web page.
+This API provides an option to trust the certificate at application level to render the page without prompting the user
+about the SSL error or can cancel the request.
+
+# Description
+
+We propose adding `ReceivingServerCertificateError` API that allows you to verify SSL certificate,  and either continue the request
+to load the resource or cancel the request.
+
+When the event is raised, WebView2 will pass a `CoreWebView2ReceivingServerCertificateErrorEventArgs` , which lets
+you to view the SSL certificate request URL, error information, inspect the certificate metadata and several options
+for responding to the SSL request.
+
+* You can verify the certificate and proceed the request if you trust.
+* You can choose to cancel the request.
+* You can choose to display default SSL interstitial page to let user to respond to the request.
+
+We also propose adding `ClearSSLDecision` API that clears certificate decisions that were added in response to proceeding with SSL certificate errors.
+
+# Examples
+
+## Win32 C++
+``` cpp
+// This example bypass default SSL interstitial page using ReceivingServerCertificateError
+// event handler and to continue with the SSL certificate that is signed by an authority
+// that WebView2 don't trust. Otherwise, cancel the request.
+void SettingsComponent::EnableCustomServerCertificateError()
+{
+  if (m_webview)
+  {
+    if (m_ServerCertificateErrorToken.value == 0)
+    {
+         CHECK_FAILURE(m_webview->add_ReceivingServerCertificateError(
+             Callback<ICoreWebView2ReceivingServerCertificateErrorEventHandler>(
+             [this](
+               ICoreWebView2* sender,
+               ICoreWebView2ReceivingServerCertificateErrorEventArgs* args) {
+               COREWEBVIEW2_WEB_ERROR_STATUS errorStatus;
+               CHECK_FAILURE(args->get_ErrorStatus(&errorStatus));
+
+               if (errorStatus == COREWEBVIEW2_WEB_ERROR_STATUS_CERTIFICATE_IS_INVALID)
+               {
+                 // Continue the request with the SSL certificate.
+                 CHECK_FAILURE(args->put_Handled(TRUE));
+               }
+               else
+               {
+                 // Cancel the request.
+                 CHECK_FAILURE(args->put_Cancel(TRUE));
+               }
+               return S_OK;
+             })
+             .Get(),
+         &m_ServerCertificateErrorToken));
+    }
+    else
+    {
+      CHECK_FAILURE(m_webview->remove_ReceivingServerCertificateError(
+        m_ServerCertificateErrorToken));
+      m_ServerCertificateErrorToken.value = 0;
+    }
+  }
+  else
+  {
+    FeatureNotAvailable();
+  }
+}
+
+// This example clears the SSL decision in response to proceeding with SSL certificate errors.
+if (m_webview)
+{
+  CHECK_FAILURE(m_webView->ClearSSLDecision());
```

Comment:
We may want to be more specific in the name about what we're clearing. We're clearing the cached decisions from the ReceivingServerCertificateError event right? What about:
* `ClearServerCertificateErrorOverrideCache` 
* `ClearIgnoredServerCertificateErrorCache`

I think it would be nice to work in the name of the event so its clear what this relates to.